job-server-api/src/spark.jobserver/NamedRddSupport.scala

```diff
@@ -6,115 +6,191 @@ import java.util.concurrent.atomic.AtomicReference
+     */
+    def get[T](name: String)(implicit timeout: Timeout = defaultTimeout): Option[RDD[T]]
 
-   * @param timeout if the RddManager doesn't respond within this timeout, an error will be thrown.
-   * @return a collection of RDD names representing RDDs managed by the RddManager.
-   */
-  def getNames()(implicit timeout: Timeout = defaultTimeout): Iterable[String]
-}
+    /**
+     * Replaces an existing RDD with a given name with a new RDD. If an old RDD for the given name existed,
+     * it is un-persisted (non-blocking) and destroyed. It is safe to call this method when there is no
+     * existing RDD with the given name. If multiple threads call this around the same time, the end result
+     * is undefined - one of the generated RDDs will win and will be returned from future calls to get().
+     *
+     * The rdd generator function will be called from the caller's thread. Note that if this is called at the
+     * same time as getOrElseCreate() for the same name, and completes before the getOrElseCreate() call,
+     * then threads waiting for the result of getOrElseCreate() will unblock with the result of this
+     * update() call. When the getOrElseCreate() succeeds, it will replace the result of this update() call.
+     *
+     * @param name the unique name of the RDD. The uniqueness is scoped to the current SparkContext.
+     * @param rddGen a 0-ary function which will be called to generate the RDD in the caller's thread.
+     * @param forceComputation if true, forces the RDD to be computed by calling count().
+     * @param storageLevel the storage level to persist the RDD with. Default: StorageLevel.MEMORY_ONLY.
+     * @tparam T the generic type of the RDD.
+     * @return the RDD with the given name.
+     */
+    def update[T](name: String,
+                  rddGen: => RDD[T],
+                  forceComputation: Boolean = true,
+                  storageLevel: StorageLevel = defaultStorageLevel): RDD[T]
+
+    /**
+     * Destroys an RDD with the given name, if one existed. Has no effect if no RDD with this name exists.
+     *
+     * @param name the unique name of the RDD. The uniqueness is scoped to the current SparkContext.
+     */
+    def destroy(name: String): Unit
+
+    /**
+     * Returns the names of all named RDDs that are managed by the RddManager.
+     *
+     * Note: this returns a snapshot of RDD names at one point in time. The caller should always expect
+     * that the data returned from this method may be stale and incorrect.
+     *
+     * @param timeout if the RddManager doesn't respond within this timeout, an error will be thrown.
+     * @return a collection of RDD names representing RDDs managed by the RddManager.
+     */
+    def getNames()(implicit timeout: Timeout = defaultTimeout): Iterable[String]
+  }
+
+  private def rddPersister[T](forceComputation: Boolean) = new NamedObjectPersister[NamedRDD[T]] {
+    override def saveToContext(namedObj: NamedRDD[T], name: String) {
```

Comment:
Ok, the reason you don't need the SparkContext here is because RDDs internally already contain a reference to the RDD.  Dunno fi that's true of everything else that the user wants persisted.